google/cloud/pubsublite/internal/wire/serial_batcher.py

```diff
@@ -54,12 +72,14 @@ def add(self, request: Request) -> "asyncio.Future[Response]":
         """
         item = WorkItem[Request, Response](request)
         self._requests.append(item)
+        self._batch_size += self._sizer.get_size(request)
         return item.response_future
 
-    def should_flush(self) -> bool:
-        return self._tester.test(item.request for item in self._requests)
+    def size(self) -> BatchSize:
```

Comment:
Most of the CPU in the publish path was in should_flush calls, which had to construct an iterable and iterate through every time.